substate-store-impls/src/substate_database_staging.rs

```diff
@@ -0,0 +1,282 @@
+use itertools::Itertools;
+use radix_engine_common::prelude::*;
+use substate_store_interface::interface::*;
+
+pub struct SubstateDatabaseStaging<'s, S> {
```

Comment:
What do you think about naming this more specifically a single layer overlay?

Reply:
Sure, I've got a few in mind, maybe:

* `SingleLayerSubstateDatabaseOverlay`
* `SingleLayerDatabaseOverlay`
* `SingleLayerDatabaseOverlay`
* `SingleOverlayStorage`

Perhaps overly verbose, but my personal pick is `SingleLayerSubstateDatabaseOverlay` :D Any preference on which one to choose?

Reply:
I went with `SingleSubstateDatabaseOverlay`. Would be happy to change to something else if you want.

Reply:
You know what, I think actually like just `SubstateDatabaseOverlay` :D.

The word overlay (instead of staging) is enough to indicate a single layer to me.